app/components/views/ProposalDetails/ProposalDetails.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState, useMemo } from "react";
+import { FormattedMessage as T } from "react-intl";
+import { InvisibleButton } from "buttons";
+import { PoliteiaLink, VerticalAccordion } from "shared";
+import {
+  ProposalNotVoting, NoTicketsVotingInfo, OverviewField, OverviewVotingProgressInfo,
+  NoElligibleTicketsVotingInfo, TimeValue, ProposalText, ProposalAbandoned
+} from "./helpers";
+import ChooseVoteOption from "./ChooseVoteOption.jsx";
+import {
+  VOTESTATUS_ACTIVEVOTE, VOTESTATUS_FINISHEDVOTE, PROPOSALSTATUS_ABANDONED
+} from "actions/GovernanceActions";
+import { useProposalDetails } from "./hooks";
+import styles from "./ProposalDetails.module.css";
+import { classNames } from "pi-ui";
+
+function ProposalDetails ({
+  viewedProposalDetails, showPurchaseTicketsPage, setVoteOption,
+  newVoteChoice, text, goBackHistory, eligibleTicketCount
+}) {
+  const {
+    creator, timestamp, endTimestamp, currentVoteChoice, hasEligibleTickets,
+    name, token, voteStatus, proposalStatus, voteOptions, voteCounts,
+    version, quorumMinimumVotes, walletEligibleTickets
+  } = viewedProposalDetails;
+  const [ showWalletEligibleTickets, toggleWalletEligibleTickets ] = useState(false);
+  const { tsDate, hasTickets } = useProposalDetails();
+
+  const voteInfo = useMemo(() => {
+    if (proposalStatus === PROPOSALSTATUS_ABANDONED) {
+      return <ProposalAbandoned />;
+    }
+    if (voteStatus === VOTESTATUS_FINISHEDVOTE) {
+      return <ChooseVoteOption {...{ voteOptions, currentVoteChoice, votingComplete: true }} />;
+    }
+    if (voteStatus === VOTESTATUS_ACTIVEVOTE) {
+      if (!hasTickets) {
+        return <NoTicketsVotingInfo {...{ showPurchaseTicketsPage }} />;
+      }
+      if (!hasEligibleTickets) {
+        return <NoElligibleTicketsVotingInfo {...{ showPurchaseTicketsPage }} />;
+      }
+      return <ChooseVoteOption {...{
+        viewedProposalDetails, voteOptions, setVoteOption, newVoteChoice,
+        eligibleTicketCount, currentVoteChoice, votingComplete: false
+      }} />;
+    }
+    return <ProposalNotVoting />;
+  }, [ proposalStatus, voteStatus, hasTickets, hasEligibleTickets, currentVoteChoice, viewedProposalDetails, eligibleTicketCount, newVoteChoice, setVoteOption, showPurchaseTicketsPage, voteOptions ]);
```

Comment:
Isn't this line too long? Also, good job memoizing this but I'd rather split it into another component and use the `React.memo()` instead. This component is quite polluted by JSX already.

Reply:
Good point mate, I've considered splitting the component here, thought maybe doing it in another PR but actually it make sense to do it at this stage when the css module are there.
I can give it a shot.

Reply:
Done mate :mate: